src/Nethermind/Nethermind.JsonRpc/JsonRpcProcessor.cs

```diff
@@ -91,46 +91,67 @@ private IEnumerable<(JsonRpcRequest Model, List<JsonRpcRequest> Collection)> Des
             {
                 if (token is JArray array)
                 {
-                    foreach (JToken tokenElement in array)
-                    {
-                        UpdateParams(tokenElement);
-                    }
-
-                    yield return (null, array.ToObject<List<JsonRpcRequest>>(_obsoleteBasicJsonSerializer));
+                    yield return (null, array.Select(JTokenToJsonRpcRequest).ToList());
                 }
                 else
                 {
-                    UpdateParams(token);
-                    yield return (token.ToObject<JsonRpcRequest>(_obsoleteBasicJsonSerializer), null);
+                    yield return (JTokenToJsonRpcRequest(token), null);
                 }
             }
         }
 
-        private void UpdateParams(JToken token)
+        /// <summary>
+        /// Manually convert JToken to JsonRpcRequests as the param inside JsonRpcRequests is a JToken. If we use
+        /// a JTokenReader, it would iterate through the JSON which is a waste of time, as we already have a JToken.
+        /// </summary>
+        /// <param name="token"></param>
+        /// <returns></returns>
+        private JsonRpcRequest JTokenToJsonRpcRequest(JToken token)
         {
-            var paramsToken = token.SelectToken("params");
-            if (paramsToken is null)
+            JsonRpcRequest jsonRpcRequest = new JsonRpcRequest();
```

Comment:
<sub>Can be just `new();` and below one more</sub>

Reply:
Done